portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -35,13 +37,17 @@
   
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    ArrayList<String> comments = new ArrayList<>();
+    // Integer guaranteed in script.js
+    int limit = Integer.parseInt(request.getParameter("limit"));
+
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
-    PreparedQuery results = this.datastore.prepare(query);
-    for (Entity entity : results.asIterable()) {
-      comments.add((String) entity.getProperty("content"));
-    }
+    PreparedQuery pq = this.datastore.prepare(query);
 
+    List<String> comments = pq.asList(FetchOptions.Builder.withLimit(limit))
+      .stream()
+      .map(entity->(String) entity.getProperty("content"))
+      .collect(Collectors.toList());
```

Comment:
Fancy

Reply:
Is this unnecessarily complicated? My intention was to use something like list comprehension in Python `comments = [(String) entity.getProperty("content") for entity in entities]` . 

Maybe I should just go with the regular way? 
```
List<String> comments;
List<Entity> entities = pq.asList(FetchOptions.Builder.withLimit(limit));
for (Entity entity : entities){
  comments.add((String) entity.getProperty("content"));
}
```

Reply:
No! This is good! Map + Array functions for simple thing are really nice.

Reply:
Haha glad to hear that!